### PR TITLE
Fix: insert/update/delete do not return until changeHandler called

### PR DESCRIPTION
**Pull request by j-berman**
### The Issue

The `changeHandler` was getting called [AFTER all transactions were applied](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/ws.js#L306-L307). This meant in a situation where a user attempts concurrent inserts, an insert could _sometimes_ resolve and return to the user BEFORE the `changeHandler` gets called.

This caused the test4 failure highlighted in PR #48

Here's the circumstance described:

1. Browser 1 inserts item foo [placing it in the array of unverified tx’s, then queues request()](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/db.js#L689-L699)
2. Browser 2 inserts item bar
3. Browser 1 receives transaction log with foo and bar and [applies this log](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/ws.js#L306)
4. Browser 1 successfully [applies and resolves foo promise, then iterates in the for loop and queues the promise to apply bar](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/db.js#L118-L131)
5. Browser 1 [receives response to request(), finishes executing the transaction and returns `insertItem(foo)` to the user](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/db.js#L702)
6. Browser 1 finishes applying bar **then finally** [calls the changeHandler with foo and bar in the state](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/ws.js#L307)

### The fix

With the approach in this PR, steps 1-3 are the same, but the latter steps look like this:

4. Browser 1 successfully applies and resolves foo promise, then [calls the changeHandler with foo in the state](https://github.com/encrypted-dev/userbase/blob/b921c1917e8e805a7532615d6d13c9c6177a4248/src/userbase-js/src/db.js#L133-L135)
5. Browser 1 receives response to request(), finishes executing the transaction and returns `insertItem(foo)` to the user
6. Browser 1 finishes applying bar then [calls the changeHandler again with bar in the state]((https://github.com/encrypted-dev/userbase/blob/b921c1917e8e805a7532615d6d13c9c6177a4248/src/userbase-js/src/db.js#L133-L135))

Note that if a transaction is successful, the [getResult() promise](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/db.js#L702) will resolve, but the changeHandler [will continue executing synchronously](https://github.com/encrypted-dev/userbase/blob/b921c1917e8e805a7532615d6d13c9c6177a4248/src/userbase-js/src/db.js#L130-L135). The original [postTransaction()](https://github.com/encrypted-dev/userbase/blob/4162cf97cca0f7a95553e62a1c0d8b01fdd89438/src/userbase-js/src/db.js#L702-L706) won't continue executing until the `changeHandler` finishes executing. 

Thus, in simpler terms, whenever a user calls `insertItem()` and it returns, the user is guaranteed to have seen the inserted item in their db change handler (same logic applies to `deleteItem()`, `updateItem()`, and `transaction()`).

#### An important note

As pointed out in a similar circumstance by @adaddeo a while back, the following circumstance is possible:
- a developer renders a loader before calling `insertItem()`
- then while the **loader** is _still_ rendered, the **item** renders because their changeHandler executes
- then `insertItem()` resolves and the developer stops rendering the loader

So the item could be rendered while the loader is still on the screen.

Note that this was the case before this PR as well. This PR simply makes the behavior consistent (the db change handler is now **guaranteed** to have been called before `insertItem()` returns, rather than **usually** getting called before `insertItem()` returns).

#### Minor note

I also got rid of the `handlerWrapper` in favor of just using the `receivedMessage` promise in a more obvious place.

### CODE REVIEW

File: src/userbase-js/src/db.js [hunk 1]
```diff
@@ -128,6 +129,14 @@ class Database {
...
       }
+
+      if (this.init && transactionCode === 'Success') {
+        this.onChange(this.getItems())
+      }
+    }
+
+    if (!this.init) {
+      this.onChange(this.getItems())
     }
```
Comment (dvassallo) [inline src/userbase-js/src/db.js]: What's the use of this.init here? And why do we call onChange from 2 places?

Comment (j-berman) [reply src/userbase-js/src/db.js]: This is an optimization that should be just as safe as only calling it in 1 place inside the for loop.

The `transactions` array passed into this `applyTransactions` function will only have >1 transactions inside it in 2 circumstances:
1. When opening a database the first time (could be a high number of transactions)
2. When the user attempts concurrent transactions (will usually be a pretty low number of transactions)

Imagine a user opens their database for the first time and they have lots of transactions in their transaction log. Without this optimization, the changeHandler would fire for every single transaction in their log.

When opening the database for the first time (i.e. `this.init` is false), we assume the client is unable to apply any other transactions aside from the first batch it receives. Considering this is our assumption inherent in the structure of the code, it's safe to wait until all transactions in the first batch are applied before firing the change handler (working on guaranteeing this assumption right now with the queue implementation we discussed).

I get that something like this might be considered pre-mature optimization, but it also allows all the tests I already have written that expect the change handler to be called when opening the database to pass. So it's an optimization that also saves me the extra work of rewriting tests :)

I was going to comment an explanation for this in the code but found it pretty complicated when typing up.

Comment (j-berman) [reply src/userbase-js/src/db.js]: If this explanation^ is satisfactory, I can write a comment in the code that has a hyperlink to it :)
